### PR TITLE
Build UI flow for consent acquiring from user

### PR DESCRIPTION
**Pull request by anshulv1401**
MOSIP-23954

1. Parsing and forwarding request URL params to the IDP server
2. QR code and otp-based authentication as a tab form. Merged from @gsasikumar's PR.
3. Authenticate service added.
4. Component for pin added.
5. consent page added.
6. login tab list updated
7. Component for Biometric authentication added.
8. L1 device discovery added.
9. Dropdown to select one among multiple devices added.
10. SBI Biometric capture integration completed.
11. Integration of configuration with OAuth-details response completed.

### CODE REVIEW

**Comment by gsasikumar** [comment]
Can we remove all the utils related images?

**Comment by anshulv1401** [comment]
> Can we remove all the utils related images?

**Comment by anshulv1401** [comment]
> > Can we remove all the utils related images?

All utils related images removed

File: src/components/L1Biometrics.js [hunk 1]
```diff
@@ -0,0 +1,322 @@
+import React, { useEffect } from "react";
+import { useState } from "react";
+import { Link, useNavigate, useSearchParams } from "react-router-dom";
+import LoadingIndicator from "../common/LoadingIndicator";
+import { challengeTypes, deviceType } from "../constants/clientConstants";
+import { ERROR, LOADED, LOADING } from "../constants/states";
+import { post_AuthenticateUser } from "../services/AuthService";
+import { encodeBase64 } from "../services/cryptoService";
```
Comment (gsasikumar) [inline src/components/L1Biometrics.js:8]: The ideal way is to inject and not import the services

File: src/components/L1Biometrics.js [hunk 2]
```diff
@@ -0,0 +1,322 @@
+import React, { useEffect } from "react";
+import { useState } from "react";
+import { Link, useNavigate, useSearchParams } from "react-router-dom";
+import LoadingIndicator from "../common/LoadingIndicator";
+import { challengeTypes, deviceType } from "../constants/clientConstants";
+import { ERROR, LOADED, LOADING } from "../constants/states";
+import { post_AuthenticateUser } from "../services/AuthService";
```
Comment (gsasikumar) [inline src/components/L1Biometrics.js:7]: We should not be importing rather we should be injecting the service or handles.

File: src/components/L1IndividualBiometric.js [hunk 3]
```diff
@@ -0,0 +1,324 @@
+import React, { useEffect } from "react";
+import { useState } from "react";
+import { Link, useNavigate, useSearchParams } from "react-router-dom";
+import LoadingIndicator from "../common/LoadingIndicator";
+import { challengeTypes, deviceType } from "../constants/clientConstants";
+import { ERROR, LOADED, LOADING } from "../constants/states";
+import { post_AuthenticateUser } from "../services/AuthService";
```
Comment (gsasikumar) [inline src/components/L1IndividualBiometric.js:7]: same comment. isolate service and components.

**Review by gsasikumar — CHANGES_REQUESTED** [review]
Some comments are generic. Can you follow that pattern

File: .env [hunk 4]
```diff
@@ -1 +1,11 @@
-REACT_APP_API_BASE_URL=http://localhost:8088/v1/idp
\ No newline at end of file
+REACT_APP_SBI_PURPOSE=Auth
```
[the remark this answers is not in the document]
Comment (anshulv1401) [reply .env]: Done

File: .env [hunk 5]
```diff
@@ -1 +1,11 @@
...
+REACT_APP_SBI_CERTIFICATION=L1
+REACT_APP_SBI_TIMEOUT=30
```
[the remark this answers is not in the document]
Comment (anshulv1401) [reply .env]: Unit for timeout is second. updated the Key to REACT_APP_SBI_TIMEOUT_IN_SECONDS

File: .env.development [hunk 6]
```diff
@@ -0,0 +1,3 @@
+REACT_APP_SBI_ENV=Staging
+REACT_APP_IDP_BASE_URL=http://localhost:8088/v1/idp
```
[the remark this answers is not in the document]
Comment (anshulv1401) [reply .env.development]: key REACT_APP_IDP_BASE_URL update to REACT_APP_IDP_API_URL

File: nginx/nginx.conf [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+
+http {
+  access_log /var/log/nginx/access1.log;
+  error_log /var/log/nginx/error1.log;
+  server {
+    listen 3000;
+    server_name  localhost;
+
+    root   /usr/share/nginx/html;
+    index  index.html index.htm;
```
[the remark this answers is not in the document]
Comment (anshulv1401) [reply nginx/nginx.conf:15]: htm is an old extention of html.

File: src/components/Consent.js [hunk 8]
```diff
@@ -0,0 +1,310 @@
+import { useEffect, useState } from "react";
+import { useSearchParams } from "react-router-dom";
+import LoadingIndicator from "../common/LoadingIndicator";
+import { otpFields } from "../constants/formFields";
```
[the remark this answers is not in the document]
Comment (anshulv1401) [reply src/components/Consent.js]: Removed

File: src/components/Consent.js [hunk 9]
```diff
@@ -0,0 +1,310 @@
...
+import { post_AuthCode } from "../services/AuthService";
+
+const fields = otpFields;
+let fieldsState = {};
+fields.forEach((field) => (fieldsState["Otp" + field.id] = ""));
```
[the remark this answers is not in the document]
Comment (anshulv1401) [reply src/components/Consent.js]: Removed

File: src/components/Consent.js [hunk 10]
```diff
@@ -0,0 +1,310 @@
+import { useEffect, useState } from "react";
+import { useSearchParams } from "react-router-dom";
+import LoadingIndicator from "../common/LoadingIndicator";
+import { otpFields } from "../constants/formFields";
+import { post_AuthCode } from "../services/AuthService";
```
[the remark this answers is not in the document]
Comment (anshulv1401) [reply src/components/Consent.js]: isolated services and components

File: src/components/L1Biometrics.js (hunk 2 above)
Comment (anshulv1401) [reply src/components/L1Biometrics.js]: isolated services and components

File: src/components/L1IndividualBiometric.js (hunk 3 above)
Comment (anshulv1401) [reply src/components/L1IndividualBiometric.js]: isolated services and components

**Review by gsasikumar — APPROVED** [review]

**Review by gsasikumar — APPROVED** [review]

**Pull request merged**